src/components/Actions.tsx

```diff
@@ -43,7 +43,9 @@ export const SelectedShapeActions = ({
         <>
           {renderAction("changeBackgroundColor")}
 
-          {renderAction("changeFillStyle")}
+          {targetElements.some(
+            (element) => !isTransparent(element.backgroundColor),
+          ) && renderAction("changeFillStyle")}
```

Comment:
This is not correct, as it appears hidden when we have 0 selected and trying to create a new rectangle for example.